app.rb

```diff
@@ -40,15 +40,30 @@
 post '/portal' do
   domain = params[:email].partition('@').last
 
-  organization = WorkOS::Portal.create_organization(
-    domains: [domain],
-    name: domain.partition('.').first,
+  organizations = WorkOS::Portal.list_organizations(
+   domains: [domain],
   )
 
-  portal_link = WorkOS::Portal.generate_link(
-    intent: 'sso',
-    organization: organization.id
-  )
+  if organizations.data.empty?
+    organization = WorkOS::Portal.create_organization(
+     domains: [domain],
+     name: domain.partition('.').first,
+    )
+
+    portal_link = WorkOS::Portal.generate_link(
+     intent: 'sso',
+     organization: organization.id
+    )
+
+    redirect portal_link
```

Comment:
If this happens regardless of the branch we could have this redirect outside of it

```rb
if organizations.data.empty?
  ...
else
  ...
end

redirect portal_link
```

local variables can escape the scope of the block they are defined in (within the closure of the method). You can also assign the return value of a block to a variable

```rb
portal_link = if organizations.data.empty?
  organization = ...

  WorkOS::Portal.generate_link(
     intent: 'sso',
     organization: organization.id
    )
else
  organization = ...

  WorkOS::Portal.generate_link(
     intent: 'sso',
     organization: organization.id
    )
end

redirect_to portal_link
```